---
created: '2021-03-18T10:37:55.960956Z'
creator: sentry
source: tests/sentry/grouping/test_variants.py
---
app-depth-1:
  hash: "59c0434e3624d23495e047f18d5615c5"
  tree_label: "log::__private_api_log"
  component:
    app-depth-1*
      exception*
        stacktrace*
          frame*
            function*
              "log::__private_api_log"
            package (ignored because function takes precedence)
              "log"
        value (ignored because stacktrace takes precedence)
          "Holy shit everything is on fire!"
--------------------------------------------------------------------------
app-depth-2:
  hash: "8cd9c779e1522f718e9270004d2029ed"
  tree_label: "log_demo::main | ... | sentry::integrations::log::Logger::log"
  component:
    app-depth-2*
      exception*
        stacktrace*
          frame*
            function*
              "log_demo::main"
            package (ignored because function takes precedence)
              "log_demo"
          frame*
            function*
              "log::__private_api_log"
            package (ignored because function takes precedence)
              "log"
          frame*
            function*
              "sentry::integrations::log::Logger::log"
        type (ignored because exception is synthetic)
          "log_demo"
        value (ignored because stacktrace takes precedence)
          "Holy shit everything is on fire!"
--------------------------------------------------------------------------
app-depth-3:
  hash: "7b1c864b21801cc624a3734761cfbc48"
  tree_label: "std::rt::lang_start::{{closure}} | log_demo::main | ... | sentry::integrations::log::Logger::log | sentry::hub::Hub::with_active"
  component:
    app-depth-3*
      exception*
        stacktrace*
          frame*
            function*
              "std::rt::lang_start::{{closure}}"
            package (ignored because function takes precedence)
              "std"
          frame*
            function*
              "log_demo::main"
            package (ignored because function takes precedence)
              "log_demo"
          frame*
            function*
              "log::__private_api_log"
            package (ignored because function takes precedence)
              "log"
          frame*
            function*
              "sentry::integrations::log::Logger::log"
          frame*
            function*
              "sentry::hub::Hub::with_active"
            package (ignored because function takes precedence)
              "sentry"
        type (ignored because exception is synthetic)
          "log_demo"
        value (ignored because stacktrace takes precedence)
          "Holy shit everything is on fire!"
--------------------------------------------------------------------------
app-depth-4:
  hash: "8484da08545a2177db5f2d6cc56ef47a"
  tree_label: "std::panicking::try::do_call | std::rt::lang_start::{{closure}} | log_demo::main | ... | sentry::integrations::log::Logger::log | sentry::hub::Hub::with_active | sentry::hub::Hub::with"
  component:
    app-depth-4*
      exception*
        stacktrace*
          frame*
            function*
              "std::panicking::try::do_call"
            package (ignored because function takes precedence)
              "std"
          frame*
            function*
              "std::rt::lang_start::{{closure}}"
            package (ignored because function takes precedence)
              "std"
          frame*
            function*
              "log_demo::main"
            package (ignored because function takes precedence)
              "log_demo"
          frame*
            function*
              "log::__private_api_log"
            package (ignored because function takes precedence)
              "log"
          frame*
            function*
              "sentry::integrations::log::Logger::log"
          frame*
            function*
              "sentry::hub::Hub::with_active"
            package (ignored because function takes precedence)
              "sentry"
          frame*
            function*
              "sentry::hub::Hub::with"
            package (ignored because function takes precedence)
              "sentry"
        type (ignored because exception is synthetic)
          "log_demo"
        value (ignored because stacktrace takes precedence)
          "Holy shit everything is on fire!"
--------------------------------------------------------------------------
app-depth-5:
  hash: "88d63e9701c93e98832eb50d7e5e9d4a"
  tree_label: "___rust_maybe_catch_panic | std::panicking::try::do_call | std::rt::lang_start::{{closure}} | log_demo::main | ... | sentry::integrations::log::Logger::log | sentry::hub::Hub::with_active | sentry::hub::Hub::with | sentry::hub::Hub::with_active::{{closure}}"
  component:
    app-depth-5*
      exception*
        stacktrace*
          frame*
            function*
              "___rust_maybe_catch_panic"
          frame*
            function*
              "std::panicking::try::do_call"
            package (ignored because function takes precedence)
              "std"
          frame*
            function*
              "std::rt::lang_start::{{closure}}"
            package (ignored because function takes precedence)
              "std"
          frame*
            function*
              "log_demo::main"
            package (ignored because function takes precedence)
              "log_demo"
          frame*
            function*
              "log::__private_api_log"
            package (ignored because function takes precedence)
              "log"
          frame*
            function*
              "sentry::integrations::log::Logger::log"
          frame*
            function*
              "sentry::hub::Hub::with_active"
            package (ignored because function takes precedence)
              "sentry"
          frame*
            function*
              "sentry::hub::Hub::with"
            package (ignored because function takes precedence)
              "sentry"
          frame*
            function*
              "sentry::hub::Hub::with_active::{{closure}}"
            package (ignored because function takes precedence)
              "sentry"
        type (ignored because exception is synthetic)
          "log_demo"
        value (ignored because stacktrace takes precedence)
          "Holy shit everything is on fire!"
--------------------------------------------------------------------------
app-depth-max:
  hash: "c2cc4110b3cf1cf35873a39525994f02"
  tree_label: "<entire stacktrace>"
  component:
    app-depth-max*
      exception*
        stacktrace*
          frame*
            function*
              "_main"
          frame*
            function*
              "std::rt::lang_start"
            package (ignored because function takes precedence)
              "std"
          frame*
            function*
              "std::rt::lang_start_internal"
            package (ignored because function takes precedence)
              "std"
          frame*
            function*
              "___rust_maybe_catch_panic"
          frame*
            function*
              "std::panicking::try::do_call"
            package (ignored because function takes precedence)
              "std"
          frame*
            function*
              "std::rt::lang_start::{{closure}}"
            package (ignored because function takes precedence)
              "std"
          frame*
            function*
              "log_demo::main"
            package (ignored because function takes precedence)
              "log_demo"
          frame*
            function*
              "log::__private_api_log"
            package (ignored because function takes precedence)
              "log"
          frame*
            function*
              "sentry::integrations::log::Logger::log"
          frame*
            function*
              "sentry::hub::Hub::with_active"
            package (ignored because function takes precedence)
              "sentry"
          frame*
            function*
              "sentry::hub::Hub::with"
            package (ignored because function takes precedence)
              "sentry"
          frame*
            function*
              "sentry::hub::Hub::with_active::{{closure}}"
            package (ignored because function takes precedence)
              "sentry"
          frame*
            function (ignored unknown function)
              "<unknown>"
            package* (used as fallback because function name is not available)
              "sentry"
          frame*
            function (ignored unknown function)
              "<redacted>"
            package* (used as fallback because function name is not available)
              "something"
        type (ignored because exception is synthetic)
          "log_demo"
        value (ignored because stacktrace takes precedence)
          "Holy shit everything is on fire!"
--------------------------------------------------------------------------
system:
  hash: "c2cc4110b3cf1cf35873a39525994f02"
  tree_label: "<entire stacktrace>"
  component:
    system*
      exception*
        stacktrace*
          frame*
            function*
              "_main"
          frame*
            function*
              "std::rt::lang_start"
            package (ignored because function takes precedence)
              "std"
          frame*
            function*
              "std::rt::lang_start_internal"
            package (ignored because function takes precedence)
              "std"
          frame*
            function*
              "___rust_maybe_catch_panic"
          frame*
            function*
              "std::panicking::try::do_call"
            package (ignored because function takes precedence)
              "std"
          frame*
            function*
              "std::rt::lang_start::{{closure}}"
            package (ignored because function takes precedence)
              "std"
          frame*
            function*
              "log_demo::main"
            package (ignored because function takes precedence)
              "log_demo"
          frame*
            function*
              "log::__private_api_log"
            package (ignored because function takes precedence)
              "log"
          frame*
            function*
              "sentry::integrations::log::Logger::log"
          frame*
            function*
              "sentry::hub::Hub::with_active"
            package (ignored because function takes precedence)
              "sentry"
          frame*
            function*
              "sentry::hub::Hub::with"
            package (ignored because function takes precedence)
              "sentry"
          frame*
            function*
              "sentry::hub::Hub::with_active::{{closure}}"
            package (ignored because function takes precedence)
              "sentry"
          frame*
            function (ignored unknown function)
              "<unknown>"
            package* (used as fallback because function name is not available)
              "sentry"
          frame*
            function (ignored unknown function)
              "<redacted>"
            package* (used as fallback because function name is not available)
              "something"
        type (ignored because exception is synthetic)
          "log_demo"
        value (ignored because stacktrace takes precedence)
          "Holy shit everything is on fire!"
